docs/architecture/maui/mvvm-community-toolkit-features.md

```diff
@@ -0,0 +1,211 @@
+
+By default, `AsyncRelayCommand` does not allow concurrent execution. This is very helpful in situations where a user could unintentionally tap a control multiple times to execute a long-running or costly operation. During task execution, `AsyncRelayCommand` will automatically call the `CanExecuteChanged` event. In .NET MAUI, controls that support the `Command` and `CommandParameter` properties, such as `Button`, will listen to this event and automatically enable or disable it during execution. This functionality can be overridden by using a custom `canExecute` parameter or setting the `AsyncRelayCommandOptions.AllowConcurrentExecutions` flag in the constructor.
+
+For more detailed information on implementing commands, see the section [Implementing commands](mvvm.md#implementing-commands) in the MVVM chapter. Detailed information for the `RelayCommand` and `AsyncRelayCommand` is available in the [Commanding](/dotnet/communitytoolkit/mvvm/relaycommand) of the MVVM Toolkit Developer Center.
+
+## Source Generators
+
+Using the MVVM Toolkit components out-of-the-box allows us to greatly simplify our ViewModel. The MVVM Toolkit allows us to simplify common code use cases even further by using [Source Generators](/dotnet/csharp/roslyn-sdk/source-generators-overview). The MVVM Toolkit source generators look for specific attributes in our code and can generate wrappers for properties and commands.
+
+> [!NOTE]
+> The MVVM Toolkit Source Generators generate code that is additive to our existing objects. Because of this, any object that is leveraging a source generator will need to be marked as `partial`.
+
+The MVVM Toolkit `ObservableProperty` attribute works on classes based-off on `ObservableObject` and will wrap a private variable as a property that generates changes. The following code shows an example of using the `ObservableObject` attribute:
+
+```csharp
+public partial class SampleViewModel : ObservableObject
+{
+    [ObservableProperty]
+    private string _name;
+}
+```
+
+With the `ObservableProperty` attribute applied to the `_name` variable, the source generator will run and generate another partial class with the following code:
```

Comment:
```suggestion
With the `ObservableProperty` attribute applied to the `_name` field, the source generator will run and generate another partial class with the following code:
```

Sure it is a variable, but so are properties, I think the important thing here is specifying that it's a field.